c2rust-analyze/src/rewrite/expr/convert.rs

```diff
@@ -232,6 +228,11 @@ fn materialize_adjustments<'tcx>(
             hir_rw
         }
         (rw @ Rewrite::Ref(..), &[Adjust::Deref(..), Adjust::Borrow(..)]) => rw,
+        (ref rw @ Rewrite::MethodCall(ref s, ..), &[Adjust::Deref(..), Adjust::Borrow(..)])
+            if s == "as_ptr" =>
+        {
+            rw.clone()
+        }
```

Comment:
Tying this to the method name `as_ptr` seems weird.  I think we could allow any method name here, since method calls will automatically reborrow (`Adjust::Deref` + `Adjust::Borrow`) when needed.

Reply:
3ebc774d